packages/babel-types/src/retrievers/getBindingIdentifiers.ts

```diff
@@ -1,20 +1,39 @@
-// @flow
 import {
   isExportDeclaration,
   isIdentifier,
   isDeclaration,
   isFunctionDeclaration,
   isFunctionExpression,
+  isExportAllDeclaration,
 } from "../validators/generated";
+import type * as t from "../types";
+
+function getBindingIdentifiers(
+  node: t.Node,
+  duplicates: true,
+  outerOnly?: boolean,
+): Record<string, Array<t.Identifier>>;
+
+function getBindingIdentifiers(
+  node: t.Node,
+  duplicates?: false,
+  outerOnly?: boolean,
+): Record<string, t.Identifier>;
+
+function getBindingIdentifiers(
+  node: t.Node,
+  duplicates?: boolean,
+  outerOnly?: boolean,
+): Record<string, t.Identifier | Array<t.Identifier>>;
```

Comment:
Q: Do we need this, or is it implied by the two above?

Also, if we need it the return type should probably be `Record<string, t.Identifier> | Record<string, Array<t.Identifier>>`.